shell/browser/browser_win.cc

```diff
@@ -105,6 +111,62 @@ base::string16 GetAppForProtocolUsingAssocQuery(const GURL& url) {
   return base::string16(out_buffer);
 }
 
+// Helper for GetApplicationInfoForProtocol().
+// takes in an assoc_str
+// (https://docs.microsoft.com/en-us/windows/win32/api/shlwapi/ne-shlwapi-assocstr)
+// and returns the application name, icon and path that handles the protocol.
+//
+// Windows 8 introduced a new protocol->executable binding system which cannot
+// be retrieved in the HKCR registry subkey method implemented below. We call
+// AssocQueryString with the new Win8-only flag ASSOCF_IS_PROTOCOL instead.
+base::string16 GetAppInfoHelperForProtocol(ASSOCSTR assoc_str,
+                                           const GURL& url) {
+  const base::string16 url_scheme = base::ASCIIToUTF16(url.scheme());
+  if (!IsValidCustomProtocol(url_scheme))
+    return base::string16();
+
+  wchar_t out_buffer[1024];
+  DWORD buffer_size = base::size(out_buffer);
+  HRESULT hr =
+      AssocQueryString(ASSOCF_IS_PROTOCOL, assoc_str, url_scheme.c_str(), NULL,
+                       out_buffer, &buffer_size);
+  if (FAILED(hr)) {
+    DLOG(WARNING) << "AssocQueryString failed!";
+    return base::string16();
+  }
+  return base::string16(out_buffer);
+}
+
+void OnIconDataAvailable(base::FilePath app_path,
+                         base::string16 app_display_name,
+                         gin_helper::Promise<gin_helper::Dictionary> promise,
+                         gfx::Image icon) {
+  if (!icon.IsEmpty()) {
+    v8::HandleScope scope(promise.isolate());
+    gin_helper::Dictionary dict =
+        gin::Dictionary::CreateEmpty(promise.isolate());
+
+    dict.Set("appPath", app_path);
+    dict.Set("displayName", app_display_name);
+    dict.Set("icon", icon);
+    promise.Resolve(dict);
+  } else {
+    promise.RejectWithErrorMessage("Failed to get file icon.");
+  }
+}
+
+base::string16 GetAppDisplayNameForProtocol(const GURL& url) {
```

Comment:
You should remove `GetAppForProtocolUsingAssocQuery` from this file and replace it's usages with this new `GetAppDisplayNameForProtocol`.  It's the same code / result right?